### PR TITLE
:seedling: Upgrade Dependencies.

### PR DESCRIPTION
**Pull request by xuezhaojun**
## Summary

Fixes: 
* https://github.com/open-cluster-management-io/ocm/pull/672
* https://github.com/open-cluster-management-io/ocm/pull/669
* https://github.com/open-cluster-management-io/ocm/pull/668

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -10,33 +10,33 @@ require (
...
 	helm.sh/helm/v3 v3.15.3
-	k8s.io/api v0.30.3
-	k8s.io/apiextensions-apiserver v0.30.3
-	k8s.io/apimachinery v0.30.3
-	k8s.io/apiserver v0.30.3
-	k8s.io/client-go v0.30.3
-	k8s.io/component-base v0.30.3
+	k8s.io/api v0.31.1
+	k8s.io/apiextensions-apiserver v0.31.1
+	k8s.io/apimachinery v0.31.2
```
Comment (qiujian16) [inline go.mod:28]: why this is 0.31.2 and others are 0.31.1 :)

Comment (xuezhaojun) [reply go.mod:28]: 0.31.1 is cause by https://github.com/open-cluster-management-io/ocm/pull/672, 0.31.2 is caused by https://github.com/open-cluster-management-io/ocm/pull/669.

I can try upgrade the other also to 0.31.2 as well.

Comment (xuezhaojun) [reply go.mod:28]: Updated all to 0.31.2.

Comment (qiujian16) [reply go.mod:28]: note we also need to upgrade addon-framekwork/sdk-go/API

**Pull request merged**